Geometry/HGCalCommonData/interface/HGCalCellUV.h

```diff
@@ -0,0 +1,48 @@
+#ifndef Geometry_HGCalCommonData_HGCalCellUV_h
+#define Geometry_HGCalCommonData_HGCalCellUV_h
+
+#include <cmath>
+#include <cstdint>
+#include <iterator>
+#include <map>
+#include "Geometry/HGCalCommonData/interface/HGCalCell.h"
+
+class HGCalCellUV {
+public:
+  HGCalCellUV(double waferSize, double separation, int32_t nFine, int32_t nCoarse);
+
+  std::pair<int32_t, int32_t> cellUVFromXY1(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY2(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY3(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+  std::pair<int32_t, int32_t> cellUVFromXY4(
+      double xloc, double yloc, int32_t placement, int32_t type, bool extend, bool debug);
+
+private:
+  std::pair<int32_t, int32_t> cellUVFromXY4(double xloc,
+                                            double yloc,
+                                            int ncell,
+                                            double cellX,
+                                            double cellY,
+                                            double cellXTotal,
+                                            double cellYTotal,
+                                            std::map<std::pair<int, int>, std::pair<double, double> >& cellPos,
+                                            bool extend,
+                                            bool debug);
+
+  const double sqrt3_ = std::sqrt(3.0);
+  const double sqrt3By2_ = (0.5 * std::sqrt(3.0));
```

Comment:
They should be rather made `static constexpr`
And I would do the same for all other constants used in the methods in the implementation file, the one that I suggest to make `constexpr` there

Reply:
Getting error "non-constexpr function 'sqrt' cannot be used in a constant expression", similarly for fcos and fsin, as they are derived from sqrt3By2_ they cannot be defined as constexpr